### PR TITLE
Preserve autograd for torch lambdify Matrix outputs

### PR DESCRIPTION
**Pull request by tmparticle**
## Summary

Preserve autograd connectivity for explicit matrix outputs in `lambdify(..., "torch")`.

## What changed

- Add `_torch_tensor(...)` in `sympy.printing.pytorch`
- Use `_torch_tensor(...)` instead of `torch.tensor(...)` in `_print_MatrixBase`
- Import `_torch_tensor` into the `"torch"` lambdify namespace
- Add a regression test showing that a matrix-valued lambdified function keeps gradients flowing

## Why

`torch.tensor([...])` reconstructs tensors from values. If any matrix entry is already a tensor that requires grad, the resulting matrix loses its autograd graph.

Using recursive `torch.stack(...)` preserves existing tensor leaves while still handling scalar constants cleanly.

## Reproducer

```python
import torch
from sympy import Matrix, symbols, sin
from sympy.utilities.lambdify import lambdify

x, y = symbols("x y")
expr = Matrix([[x, sin(y)], [1, 2]])
f = lambdify([x, y], expr, "torch")

x_val = torch.tensor(1.0, dtype=torch.float64, requires_grad=True)
y_val = torch.tensor(2.0, dtype=torch.float64, requires_grad=True)
result = f(x_val, y_val)
result.sum().backward()
```

After this patch, `result.requires_grad` is true and gradients flow back to both inputs.

## Validation

- Minimal local reproducer:
  - `result.requires_grad == True`
  - `result.grad_fn` is `StackBackward0`
  - `x_val.grad == 1.0`
  - `y_val.grad == cos(2.0)`
- Targeted pytest:
  - `MPLCONFIGDIR=/tmp/mpl PYTHONPATH=/tmp/sympy-test-deps:/tmp/sympy-torch-stack-pr /Users/tmori/work/tmpmono/.venv/bin/python -m pytest sympy/printing/tests/test_torch.py -q`
  - `16 passed in 1.06s`

#### AI Generation Disclosure

OpenAI Codex assisted with drafting the patch and PR text. The code changes were reviewed and validated locally before submission.

#### Release Notes

* printing
  * Fixed `lambdify(..., "torch")` so matrix-valued outputs preserve PyTorch autograd connectivity.

Closes #29464

### CODE REVIEW

**Comment by tmparticle** [comment]
Follow-up validation: after installing \ locally, I ran the targeted torch printer test file with the patched checkout:\n\n\
no tests ran in 0.34s\n\nResult: \.

**Comment by tmparticle** [comment]
Follow-up validation: after installing `hypothesis` locally, I ran the targeted torch printer test file with the patched checkout:

`MPLCONFIGDIR=/tmp/mpl PYTHONPATH=/tmp/sympy-test-deps:/tmp/sympy-torch-stack-pr /Users/tmori/work/tmpmono/.venv/bin/python -m pytest sympy/printing/tests/test_torch.py -q`

Result: `16 passed in 1.06s`.
